src/AnalyzeView/GeoTagController.cc

```diff
@@ -73,15 +76,191 @@ void GeoTagWorker::run(void)
     _cancel = false;
     emit progressChanged(0);
 
+    //used to time operations to get a feel for how much to progress the progressBar
+    QElapsedTimer timerTotal;
+    QElapsedTimer timerLoadImages;
+    QElapsedTimer timerParseExif;
+    QElapsedTimer timerFilter;
+    QElapsedTimer timerLoadLogFile;
+    QElapsedTimer timerGeotag;
+
+    timerTotal.start();
+
+
+    //////////// Load Images
+    timerLoadImages.start();
+
+    QDir imageDirectory = QDir(_imageDirectory);
+    if(!imageDirectory.exists()) {
+        emit error(tr("Cannot find the image directory"));
+        return;
+    }
+    if(!imageDirectory.mkdir(_imageDirectory + "/TAGGED")) {
+        emit error(tr("Images have already been tagged"));
+        return;
+    }
+
+    imageDirectory.setFilter(QDir::Files | QDir::Readable | QDir::NoSymLinks | QDir::Writable);
+    imageDirectory.setSorting(QDir::Name);
+    QStringList nameFilters;
+    nameFilters << "*.jpg" << "*.JPG";
+    imageDirectory.setNameFilters(nameFilters);
+
+    QFileInfoList imageList = imageDirectory.entryInfoList();
+    if(imageList.isEmpty()) {
+        emit error(tr("The image directory doesn't contain images, make sure your images are of the JPG format"));
+        return;
+    }
+
+    _imageBuffers.clear();
+    for (int i = 0; i < imageList.size(); ++i) {
+        QFile file(imageList.at(i).absoluteFilePath());
+        if (!file.open(QIODevice::ReadOnly)) {
+            continue;
+        }
+        _imageBuffers.append(QByteArray(file.readAll()));
+        file.close();
+    }
+
+    qWarning() << "Image loading time elapsed: " << timerLoadImages.elapsed() << " milliseconds";
```

Comment:
These shouldn't be warnings. They should be qCDebug with a new logging category for geotagging.